### ISSUE TITLE
AD0001: Analyzer 'Vogen.Rules.AddValidationAnalyzer' threw an exception

### ISSUE THREAD

**Issue by CheloXL** [SteveDunn/Vogen#450]
### Describe the bug

Using Rider, I'm getting the following error as a "warning" message:

```
AD0001: Analyzer 'Vogen.Rules.AddValidationAnalyzer' threw an exception of type 'System.NullReferenceException' with message 'Object reference not set to an instance of an object.'. 
Exception occurred with following context: Compilation: Blinkay.Control ISymbol: ParkingRightId (NamedType)  System.NullReferenceException: Object reference not set to an instance of an object. at Vogen.Rules.AddValidationAnalyzer.AnalyzeSymbol(SymbolAnalysisContext context) in /_/src/Vogen/Rules/AddValidationAnalyzer.cs:line 72 at Microsoft.CodeAnalysis.Diagnostics.AnalyzerExecutor.c. b__48_1(ValueTuple`2 data) in Z:\BuildAgent\work\3b7ce003563d6f8f\src\Compilers\Core\Portable\DiagnosticAnalyzer\AnalyzerExecutor.cs:line 448 at Microsoft.CodeAnalysis.Diagnostics.AnalyzerExecutor.ExecuteAndCatchIfThrows_NoLock[TArg](DiagnosticAnalyzer analyzer, Action`1 analyze, TArg argument, Nullable`1 info) in Z:\BuildAgent\work\3b7ce003563d6f8f\src\Compilers\Core\Portable\DiagnosticAnalyzer\AnalyzerExecutor.cs:line 1199 -----
```

There is nothing strange in the ValueObject, defined as following:
```cs
[ValueObject]
public sealed partial class ParkingRightId : IValidEntityId
{
	public bool IsValid
	{
		get { return _isInitialized && !string.IsNullOrEmpty(_value); }
	}
}
```

The solution is quite complex, so I don't think I can provide a simplified sample.

### Steps to reproduce

![image](https://github.com/SteveDunn/Vogen/assets/201414/d71e8872-6732-4bf6-9798-ea30f80800e2)

### Expected behaviour

No warning should be set, nor any error should occur.

**Comment by SteveDunn** [comment]
Thanks for reporting, I'll try to get this fixed ASAP! A few questions:
1. Do you have any global config for Vogen defaults? I can see from the attribute in your example that you default the underlying type to `string` (it's normally an `int`)
2. I take it `IsValid` is part of `IValidEntityId`?
3. Is there anything in particular that you're doing when you see this? Maybe building, or editing, or a nuget restore?

**Comment by CheloXL** [comment]
1. Yes: 
```csharp
[assembly: VogenDefaults(
	typeof(string),
	Conversions.TypeConverter | Conversions.SystemTextJson | Conversions.EfCoreValueConverter,
	customizations: Customizations.TreatNumberAsStringInSystemTextJson,
	debuggerAttributes: DebuggerAttributeGeneration.Basic
)]
```

2. Correct
3. No. Just opening the solution gives me the error.

**Comment by SteveDunn** [comment]
Thanks, I'm sure I'll spot it if it does occur on my Rider. I'm using 2023.1.3 (Build #RD-231.9161.46, built on June 21, 2023); where do you see this `AD0001` error pop up?

**Comment by SteveDunn** [comment]
There's obviously something wrong with that analyzer, as it's supposed to show this, but it only shows it when the type is on the attribute (and not in global config)

![image](https://github.com/SteveDunn/Vogen/assets/263416/8544300e-067f-46a3-a38b-98d329f0438a)


I'll continue looking!

**Comment by CheloXL** [comment]
I'm using 2023.2 EAP 9. EAP has a lot more information when source generators fail. In fact, it has a special debugger for source generators (of course you must have access to the source of the source generator and know what are you doing... a thing I'm not qualified in)

**Comment by SteveDunn** [comment]
Getting somewhere - I now have an integration test that generates the error you're seeing:

```
Expected diagnostic:
    // /0/Test0.cs(20,26,20,34): info AddValidationMethod: Type name 'TypeName' can have validation
VerifyCS.Diagnostic().WithSpan(20, 26, 20, 34).WithArguments("TypeName"),

Actual diagnostic:
    // error AD0001: Analyzer 'Vogen.Rules.AddValidationAnalyzer' threw an exception of type 'System.NullReferenceException' with message 'Object reference not set to an instance of an object.'.
```

**Comment by SteveDunn** [comment]
This is fixed in version 3.0.20 - just pushing to NuGet now. Thanks very much for the bug report - much appreciated!
